catalog/resource_sql_table.go

```diff
@@ -0,0 +1,259 @@
+package catalog
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+	clustersApi "github.com/databricks/databricks-sdk-go/service/clusters"
+	"github.com/databricks/terraform-provider-databricks/clusters"
+	"github.com/databricks/terraform-provider-databricks/common"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type SqlColumnInfo struct {
+	Name             string `json:"name"`
+	TypeText         string `json:"type_text"`
+	TypeJson         string `json:"type_json,omitempty"`
+	TypeName         string `json:"type_name"`
+	TypePrecision    int32  `json:"type_precision,omitempty"`
+	TypeScale        int32  `json:"type_scale,omitempty"`
+	TypeIntervalType string `json:"type_interval_type,omitempty"`
+	Position         int32  `json:"position"`
+	Comment          string `json:"comment,omitempty"`
+	Nullable         bool   `json:"nullable,omitempty" tf:"default:true"`
+	PartitionIndex   int32  `json:"partition_index,omitempty"`
+}
+
+type SqlTableInfo struct {
+	Name                  string            `json:"name"`
+	CatalogName           string            `json:"catalog_name" tf:"force_new"`
+	SchemaName            string            `json:"schema_name" tf:"force_new"`
+	TableType             string            `json:"table_type" tf:"force_new"`
+	DataSourceFormat      string            `json:"data_source_format"`
+	ColumnInfos           []SqlColumnInfo   `json:"columns,omitempty" tf:"alias:column,computed"`
+	StorageLocation       string            `json:"storage_location,omitempty" tf:"suppress_diff"`
+	StorageCredentialName string            `json:"storage_credential_name,omitempty" tf:"force_new"`
+	ViewDefinition        string            `json:"view_definition,omitempty"`
+	Owner                 string            `json:"owner,omitempty" tf:"computed"`
+	Comment               string            `json:"comment,omitempty"`
+	Properties            map[string]string `json:"properties,omitempty"`
+	ClusterID             string            `json:"cluster_id,omitempty" tf:"computed"`
+
+	exec common.CommandExecutor
+}
+
+type SqlTablesAPI struct {
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+func NewSqlTablesAPI(ctx context.Context, m any) SqlTablesAPI {
+	return SqlTablesAPI{m.(*common.DatabricksClient), context.WithValue(ctx, common.Api, common.API_2_1)}
+}
+
+func (a SqlTablesAPI) getTable(name string) (ti SqlTableInfo, err error) {
+	err = a.client.Get(a.context, "/unity-catalog/tables/"+name, nil, &ti)
+	return
+}
+
+func (a SqlTablesAPI) deleteTable(name string) error {
```

Comment:
SDK - https://github.com/databricks/databricks-sdk-go/blob/main/service/unitycatalog/api.go#L1852

Reply:
We decided internally to keep this model local until the unity API has full support for tables